### ISSUE TITLE
[bug] Mobile: Opens iOS App Store MetaMask page even Metamask App is installed

### ISSUE THREAD

**Issue by bichenkk** [rainbow-me/rainbowkit#956]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### RainbowKit Version

0.8.1

### wagmi Version

0.10.1

### Current Behavior

I was testing on iOS Safari browser.

When I click MetaMask connect, it sometimes successfully goes to the MetaMask App and it sometimes goes to the App Store page.

### Expected Behavior

When MetaMask App is not installed, deep link goes to the App Store MetaMask page.
When MetaMask App is installed, deep link goes to the MetaMask App (with actions).

### Steps To Reproduce

Open on Mobile and connect MetaMask.

**Comment by DanielSinclair** [comment]
Hey @bichenkk! This is one of the toughest issues around WalletConnect for us. It stems from limitations on the way the App Link scheme works on iOS. This issue is actually specifically related to MetaMask's configuration of WalletConnect, rather than the RainbowKit project. We've worked to solve this for Rainbow users with a fallback page instead of the App Store, but MetaMask hasn't implemented something similar. The WalletConnect team is exploring improvements in this area. Have a look at some documentation we contributed to the WalletConnect project that goes into greater detail: https://docs.walletconnect.com/1.0/mobile-linking#ios-app-link-constraints
